nemo/collections/common/tokenizers/bytelevel_tokenizers.py

```diff
@@ -38,9 +36,20 @@ def normalize(self, text) -> str:
 
 
 class ByteLevelTokenizer(TokenizerSpec):
-    def __init__(self):
+    def __init__(self, special_tokens: Optional[Union[Dict[str, str], List[str]]] = None):
         self.vocab_size = 259
-        self.special_tokens = [self.bos_id, self.eos_id, self.pad_id]
+        self.special_start = 256
+        self.special_token_to_id = {
+            self.pad_id: self.pad_id,
+            self.bos_id: self.bos_id,
+            self.eos_id: self.eos_id,
+        }
+        special_tokens = {} if special_tokens is None else special_tokens
+        for tok in special_tokens:
+            self.special_start -= 1
```

Comment:
Why do we count backwards here? Shouldn't it be forward?

Reply:
This is done to akin to how ByT5 handles sentinels. So we don't increase the vocab, but resue starting from the last

Reply:
I could not find it in the paper. Is that how they implemented it?
Wouldn't that mean that some input bytes will be treated as a special character?

Reply:
ok let's keep it as the paper, for sake of compatibility.